lib/kafka-consumer.js

```diff
@@ -413,6 +413,52 @@ KafkaConsumer.prototype.unsubscribe = function() {
   return this;
 };
 
+/**
+ * Read a number of messages from a specific topic and partition.
+ *
+ * Can be useful if the consume performance differs per partition. Consuming
+ * per partition could prevent slow performance on one partition from affecting
+ * the consumption of other partitions.
+ *
+ * To select the right partition it is required to set a topic param, because a
+ * consumer can be subscribed to multiple topics.
+ *
+ * @param {string} topic - Name of topic to read
+ * @param {number} partition - Identifier of partition to read
+ * @param {KafkaConsumer~readCallback} cb - Callback to return when work is done.
+ * @param {object} options - Optional settings for consume
+ */
+ KafkaConsumer.prototype.consumeFromToppar = function(topic, partition, cb, options = {}) {
+  if(typeof topic !== 'string') {
+    throw new TypeError('Topic must be a string');
+  }
+  if(typeof partition !== 'number') {
+    throw new TypeError('Partition must be a number');
+  }
+  if(typeof cb !== 'function') {
+    throw new TypeError('Callback must be a function');
+  }
+  if(typeof options !== 'object') {
+    throw new TypeError('Options must be an object');
+  }
+  if(typeof options.numberOfMessages !== 'number' && typeof options.numberOfMessages !== 'undefined') {
+    throw new TypeError('options.numberOfMessages must be a number or undefined');
+  }
+  if(typeof options.timeoutMs !== 'number' && typeof options.timeoutMs !== 'undefined') {
+    throw new TypeError('options.timeoutMs must be a number or undefined');
+  }
+  if(typeof options.onlyApplyTimeoutToFirstMessage !== 'boolean' && typeof options.onlyApplyTimeoutToFirstMessage !== 'undefined') {
+    throw new TypeError('options.onlyApplyTimeoutToFirstMessage must be a boolean or undefined');
+  }
+
+  var numberOfMessages = options.numberOfMessages ?? 1;
```

Comment:
Usage of `??` requires node 14, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing. I'm not sure they are willing to accept that requirement when we try to get it back into node-rdkafka.

Reply:
In v3 of the lib they support [node >= 16](https://github.com/Blizzard/node-rdkafka/blob/master/package.json#L46), so this should be fine.